packages/runtime/container-runtime/src/summaryGenerator.ts

```diff
@@ -0,0 +1,395 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger, ITelemetryProperties } from "@fluidframework/common-definitions";
+import { Deferred, IPromiseTimer, IPromiseTimerResult, Timer } from "@fluidframework/common-utils";
+import { ISummaryConfiguration, MessageType } from "@fluidframework/protocol-definitions";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import {
+    GenerateSummaryResult,
+    IGenerateSummaryOptions,
+    ISummarizerInternalsProvider,
+} from "./summarizerTypes";
+import { IClientSummaryWatcher } from "./summaryCollection";
+
+export const checkNotTimeout = <T>(something: T | IPromiseTimerResult | undefined): something is T => {
```

Comment:
This is checking that timeout did not happen right? I am curious as to why not the other way around - checkTimeout? Is it because there are more instances of checks for when a timeout does not happen?

Reply:
It's just this way to abuse the `something is T` return value. It really should be cleaned up.